Sources/Authentication/Credentials/Token.swift

```diff
@@ -0,0 +1,55 @@
+// MARK: Data structure
+
+public struct Token: Crendentials {
+    public let string: String
+
+    public init(string: String) {
+        self.string = string
+    }
+}
+
+// MARK: Authenticatable
+
+public protocol TokenAuthenticatable: Authenticatable {
+    /// The token entity that contains a foreign key
+    /// pointer to the user table
+    associatedtype TokenType: TokenProtocol
+
+    /// Returns the user matching the supplied token.
+    static func authenticate(_: Token) throws -> Self
+}
+
+public protocol TokenProtocol {
```

Comment:
I don't like this name, it makes it feel like it's a protocol for representing a Token, but it's way more about fetching. Maybe TokenAuthenticator?

Is it possible to merge `TokenProtocol` and `TokenAuthenticatable`?

Otherwise, maybe just `tokenAuthorizationProtocol`